packages/serverless-orchestration/src/ServerlessHub.js

```diff
@@ -40,7 +40,14 @@ const { Logger, delay } = require("@uma/financial-templates-lib");
 let logger;
 let spokeUrl;
 let customNodeUrl;
-let hubConfig = { configRetrieval: "localStorage", saveQueriedBlock: "localStorage", spokeRunner: "localStorage" };
+let hubConfig = {};
+
+const defaultHubConfig = {
+  configRetrieval: "localStorage",
+  saveQueriedBlock: "localStorage",
+  spokeRunner: "localStorage",
+  rejectSpokeDelay: 60 // 1 min.
```

Comment:
Is there any reason not to bump this default to 5 minutes to be a bit more lenient? I'm thinking especially about the case when we're using our own nodes that aren't hyper-optimized like infura. Obviously, anything longer than 5 minutes feels a little problematic because it means the next bot iteration starts before the last finishes.

Reply:
I chose this because a timed out spoke will be re-tried. Therefore the max hub execution time is ~2x the `rejectSpokeDelay`. Looking at past loops within our serverless infrastructure, the hub normally hears back from ALL spokes and resolves all serverless calls within 30 seconds from the initialization of the loop. From this information, I have set the default timeout to 60 seconds per spoke. This means that if a spoke times out it should be caught and re-tried after 60 seconds.


You are right that this heuristic only works within the past infura calls and does not work well with anything longer. However, I feel that at an upper bound the `rejectSpokeDelay` should be no more than 1/2 the execution delay of the hub. So the hub runs every 5 mins this value should be strictly less than 2.5 mins. The reason is that I really don't want the previous hub execution still running when the next one starts or it is very difficult to reason about the logs.

The highest upper bound I think we should go for this value is 2 mins. Processes _really_ should not take longer than this and if they are then we need to speed them up pretty drastically! 

WDYT?

Reply:
Agreed we can't set this too close to the POLLING_DELAY otherwise the kill signal might not be sent before the next loop starts. 2-2.5 mins seems right for an upper bound.

Reply:
Agreed @nicholaspai. I have set it to 2 mins by default, making the max hub execution time ~4mins.